builder/optimizations.cc

```diff
@@ -836,7 +837,9 @@ class in_place_aliaser : public stmt_mutator {
     } else if (can_alias && fwd && fwd->defined() && buffers.lookup(*fwd) &&
                fold_factors_strides_same(op->dims, buffers[*fwd]->dims)) {
       backward.erase(*fwd);
-      set_result(clone_buffer::make(op->sym, *fwd, std::move(body)));
+      var inplace_sym = ctx.insert_unique(ctx.name(*fwd) + ".inplace");
+      stmt cropped = crop_buffer::make(op->sym, inplace_sym, dims_bounds(op->dims), std::move(body));
+      set_result(clone_buffer::make(inplace_sym, *fwd, std::move(cropped)));
```

Comment:
Would it make sense to shadow the original sym here instead of making a new one?

Reply:
That would be perfect, but I was afraid it might break simplifier or something after.

Reply:
We deshadow again before the next simplify. I think it is probably OK. We do this in other places too.

Reply:
Huh, totally missed that, will fix shortly.

Reply:
Done.